docs/sphinx/developer/components/branchPred.rst

```diff
@@ -17,13 +17,15 @@ The usage of these parameters within a branch predictor's ``predict`` function i
 
 The ``update`` function is passed the branch outcome, the instruction address, and the branch type. From this information, any algorithms or branch structures may be updated.
 
+The state of the branch predictor when ``predict`` is called on a branch is stored in the ``ftq`` to be used by the ``update`` function.  The ``ftq`` is a queue that has an entry for each in-flight branch.  A single entry is added to the back of the ftq on ``predict``, and a single entry is removed from the front of the queue on ``update`` and from the back of the queue on ``flush``.
+
 Generic Predictor
 -----------------
 
 The algorithm(s) held within a ``BranchPredictor`` class instance can be model-specific, however, SimEng provides a ``GenericPredictor`` which contains the following logic.
 
 Global History
-    For indexing relevant prediction structures, a global history can be utilised. The global history value uses n-bits to store the n most recent branch direction outcomes, with the left-most bit being the oldest.
+    For indexing relevant prediction structures, a global history can be utilised. The global history value stores the n most recent branch direction outcomes as a bitmap, with the least-significant bit being the most recent. The global history is speculatively updated on ``predict``, and is corrected if needed on ``update`` and ``flush``.  To facilitate this speculative updating, and rolling-back on correction, for a global history of n the branch predictor keeps track of the 2n most recent branch outcomes.
```

Comment:
Maybe the wrong nomenclature by me.  I've updated

Reply:
The update is better. If we are being REALLY pedantic though, the size of unsigned int is architecture dependent but usually 32 bits rather than what is actually used which is a uint64_t which is obviously 64 bits. This would have implications on the maximum size of the history e.g. 16 vs 32 and so could cause confusion. Either use some variation of "64 bit unsigned int/uint64_t" or keep it generic with "unsigned integer". The latter especially, but in both cases the range of valid values should be provided. Providing ranges for all parameters in the docs has been discussed in the recent past so may as well start here

Reply:
Updated to specify valid history lengths

Reply:
Would the valid range (still allowing for keeping track of 2n most recent branches) not be 1-32? We can allow up to 64 bits but won't be able to store 128 predictions for speculative updating and so contradicts the lines above.

I don't think this aligns with the model config limitations which allows for a length up to UINT16_MAX - this should be updated to 32 (or 64)

Reply:
The update should be applied below in the perceptron predictor section also